### PR TITLE
Implement OSHI to obtain CPU and memory info

### PR DESCRIPTION
**Pull request by valib**
This CPU info could be used for https://github.com/yakka34/Unofficial-cpubenchmark.net-API offered by @Sami32 to obtain the CPU score discussed in https://github.com/UniversalMediaServer/UniversalMediaServer/commit/70914a8c03a455d02dc6b9fbc883638cf6c08f10

---

This change is [Reviewable](https://reviewable.io/reviews/universalmediaserver/universalmediaserver/979)

### CODE REVIEW

**Comment by Sami32** [comment]
@valib That's great :+1:  Thank you very much for your work :-)

**Comment by SubJunk** [comment]
Nice work

**Comment by SubJunk** [comment]
Happy for this to be merged if you think it's ready @valib

**Comment by valib** [comment]
@SubJunk now it could be merged.

**Comment by SubJunk** [comment]
@valib great, I'll test on OS X

File: src/main/java/net/pms/PMS.java [hunk 1]
```diff
@@ -1368,18 +1375,46 @@ public static int getOSBitness() {
...
+		return cpuScore;
+	}
+
 	/**
 	 * Log system properties identifying Java, the OS and encoding and log
 	 * warnings where appropriate.
 	 */
 	private void logSystemInfo() {
 		long memoryInMB = Runtime.getRuntime().maxMemory() / 1048576;
+		HardwareAbstractionLayer hal = new SystemInfo().getHardware();
```
Comment (SubJunk) [inline src/main/java/net/pms/PMS.java]: We should probably wrap this in getExternalNetwork(), and feel free to rename that to isExternalNetwork too, or I can

File: src/main/java/net/pms/PMS.java [hunk 2]
```diff
@@ -1368,18 +1375,46 @@ public static int getOSBitness() {
...
+        	
+        }
 
 		LOGGER.info("Java: " + System.getProperty("java.vm.name") + " " + System.getProperty("java.version") + " " + System.getProperty("sun.arch.data.model") + "-bit" + " by " + System.getProperty("java.vendor"));
 		LOGGER.info("OS: " + System.getProperty("os.name") + " " + getOSBitness() + "-bit " + System.getProperty("os.version"));
-		LOGGER.info("Encoding: " + System.getProperty("file.encoding"));
-		LOGGER.info("Memory: " + memoryInMB + " " + Messages.getString("StatusTab.12"));
+		LOGGER.info("PC total physical memory: " + memory.getTotal() / 1048576 + " " + Messages.getString("StatusTab.12"));
+		LOGGER.info("PC physical memory currently available: " + memory.getAvailable() / 1048576 + " " + Messages.getString("StatusTab.12"));
+		LOGGER.info("Max amount of memory accesible by Java: " + memoryInMB + " " + Messages.getString("StatusTab.12"));
```
Comment (SubJunk) [inline src/main/java/net/pms/PMS.java]: *accessible

File: src/main/java/net/pms/PMS.java [hunk 3]
```diff
@@ -1368,18 +1375,46 @@ public static int getOSBitness() {
...
 	 */
 	private void logSystemInfo() {
 		long memoryInMB = Runtime.getRuntime().maxMemory() / 1048576;
+		HardwareAbstractionLayer hal = new SystemInfo().getHardware();
+        CentralProcessor processor = hal.getProcessor();
+        GlobalMemory memory = hal.getMemory();
+        String name = processor.getName();
+        String result = CbmAPI.getCpuByName(name);
+        String score = "not available";
+        if (result.contains("Score")) {
```
Comment (SubJunk) [inline src/main/java/net/pms/PMS.java]: It added a noticeable lag to the program startup for me so I think we should store the result somewhere and use that every subsequent time until the CPU changes

**Comment by Sami32** [comment]
@valib It's not running on Windows XP, i mean UMS doesn't start.
If i do a clean install, i can get the interface asking to select the language for the interface but after that the _javaw.exe_ process is closed.

**Comment by valib** [comment]
@Sami32 I will test it at my work where I have the really old PC.

**Comment by Sami32** [comment]
@valib I will test it with your new commit, and wait much longer...perhaps i didn't.

Don't misunderstand me, i'm not asking that work for me, i just report it.
So don't bother you, if it's work fork fine for others users it's fine for me, don't worry...
I think i get your point, no more report, right ?

**Comment by valib** [comment]
@Sami32 please continue I like your reports :smile: but there is a fundamental question how long we will take care of the no longer supported OS and Java versions,

**Comment by Sami32** [comment]
@valib Mac OS X users using unicode files still need Java 6 version, MediaInfo bug though...but for now they still need.

The answer is up to you: https://nakedsecurity.sophos.com/2016/04/11/millions-of-people-are-still-running-windows-xp/
We don't have UMS users statistics though...but i do understand you.

**Comment by Nadahar** [comment]
In my opinion removing XP support is way premature. M$ is trying to make it as difficult as possible for people to still run XP, there's no reason to help them. The US military and many big companies will be running XP for years to come, there's nothing wrong with it - M$ basicly want to force people to buy a new OS from them. I still have XP on several computers, although I have more Windowns 7's than XP's now. It's not because the hardware can't run a newer OS, it's simply because there should be no reason to be forced to cash out for a new OS when XP is doing its job perfectly well.

Windows XP is still the third most used OS, so all those discontinuing its support is really just shooting their own foot. Let's face it, the big great lift was from Windows 4 to Windows 5. Windows 6-10 has just been thinkering and GUI redesigns for Windows 5. That's the reality that M$ is so desperatly trying to obscure.

The day there's a good reason to discontinue XP, that there's some major issue that's not possible in XP, I'd give the thought a good run, but when there's just things like MediaInfo running M$' errands it shouldn't be much of a discussion. The installer could fix that by installing the last working version if the OS was < Vista.

**Comment by SubJunk** [comment]
@Nadahar that's a good idea about installing the last working version. It could be a cool strategy to be used for Java updates too - if you have Java 6 installed and attempt to install a Java 7 version, then the last Java 6 version is downloaded for you.

I agree about continuing to support old OSes until there's good reason not to

**Comment by Nadahar** [comment]
@valib What happened with the commit message in  c74e98b?

**Comment by valib** [comment]
@Nadahar actually I don't know what it means. What is strange I can't run this PR because of the error during building it. 
     INFO  00:52:10.253 [net.pms.PMS.main()] Build: ${git.com (${git.commit.time})
ERROR 00:52:12.270 [net.pms.PMS.main()] A serious error occurred during Universal Media Server initialization: For input string: "Low"
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS

That is why I merged the latest master into this PR to ask others to test it. You were faster then me writing the comment.

**Comment by Nadahar** [comment]
I'm not that sure about the whole idea behind this - measuring the capability of the CPU and use that to decide what to transcode into. It's very biased from the point of view of those with slow/bad networks. 

As mentioned before, this idea assumes that transcoding is the only thing that CPU is doing. Computers can do multiple things, and virtualization is very common these days. Why waste CPU cycles on something that's not needed (if the network is of quality)? Why starve other VMs of resources for no good reason, and why waste power consumption and component wear (electronics wear faster with higher temperatures, fans and pumps have to run faster) for no good reason? This only makes sense for those with a poor network and loads of *spare* CPU resources.

I have plenty of CPU resources and 1 GBps network, so either would work for me - but I want the best quality I can get as "cheap" as possible. Also I have a hard time believing that a higher lossy compression doesn't lead to actual quality loss despite new and "smart" standards that tries to compensate for various effects. Sure, if that's all your network can deliver it is preferrable, but for everybody else - why reduce their quality? Personally I would prefer of video was transcoded into a lossless format, but I guess that's not really an option within DLNA? In any case, my point is that this idea just takes one infrastructure scenario into consideration and pretty much leaves everyone with different infrastructure suffering.

**Comment by valib** [comment]
I can't find what is wrong with this PR so I close it. Maybe it is related to the WIN version.

**Comment by Nadahar** [comment]
@valib I'm in the middle of checking it out for a quick look. I don't have time for anything deep now though.

**Comment by valib** [comment]
@Nadahar thanks for helping me.

**Comment by Sami32** [comment]
@Nadahar I don't know what will be the problem if we revert to JNA 4.3.0 ? None from what i'm aware.

I think he planned to update in the next version, if i remember correctly.

**Comment by Nadahar** [comment]
@Sami32 I don't know why you comment on this here. You must have a very short memory, because not long ago we had another problem where you only could get JNA 4.4.0 to work...

**Comment by Sami32** [comment]
@Nadahar He he...who have a short memory this time :wink: 
You JNA test worked fine with JNA 4.3.0 as well.

**Comment by Nadahar** [comment]
@Sami32 I can't find it now (I don't remember where we had this discussion), but it was about a week ago when you talked about the "sabotage"/"malware" on my part. The way I understood you then, it didn't work with JNA 4.3.0 but with JNA 4.4.0.

**Comment by Sami32** [comment]
"sabotage" yes ;-) but "malware" it's only in your nightmare :P
I finally found where it was:
https://github.com/UniversalMediaServer/UniversalMediaServer/issues/1303

> JNa 4.3.0 and 4.4.0 work very well on Windows XP SP3 32-bit, under Java 7u131.

**Comment by Sami32** [comment]
@Nadahar None, as you saw. That's said i was sure that you'll said that. Guy of bad faith :tongue: 

Yes, waiting doesn't really matter, as i don't think many users will be in that situation.

**Pull request closed without merging**